### PR TITLE
[CLOUDTRUST-1695] Reset SmsSent attribute + add it in the user repr

### PR DESCRIPTION
**Pull request by bsoniam**

### CODE REVIEW

File: pkg/management/component.go [hunk 1]
```diff
@@ -633,6 +635,35 @@ func (c *component) SendReminderEmail(ctx context.Context, realmName string, use
...
+	if err != nil {
+		c.logger.Warn(ctx, "err", err.Error())
+		return err
+	}
+
+	//reset the counter, if the smsSent attribute exists
+	resetCounter := 0
+	if userKc.Attributes != nil {
+		var m = *userKc.Attributes
+		if m["smsSent"] != nil {
```
Comment (fperot74) [inline pkg/management/component.go]: As an option, we could avoid calling kc.UpdateUser is m["smsSent"] is nil or equal to 0 (but not an issue)

File: pkg/management/component_test.go [hunk 2]
```diff
@@ -1699,6 +1699,57 @@ func TestSendReminderEmail(t *testing.T) {
...
+
+		err := managementComponent.ResetSmsCounter(ctx, "master", userID)
+
+		assert.Nil(t, err)
+	}
+
+	// Error at GetUser
+	{
+
+		mockKeycloakClient.EXPECT().GetUser(accessToken, realmName, userID).Return(kc.UserRepresentation{}, fmt.Errorf("error")).Times(1)
```
Comment (fperot74) [inline pkg/management/component_test.go]: You also can write a test when m["smsSent"] is already set (with any value)

Comment (bsoniam) [reply pkg/management/component_test.go]: Done

File: pkg/management/component.go (hunk 1 above)
Comment (bsoniam) [reply pkg/management/component.go]: Done

**Pull request merged**